liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/InstallAppMojoSupport.java

```diff
@@ -33,24 +35,34 @@
  * Install artifact into Liberty server support.
  */
 public class InstallAppMojoSupport extends PluginConfigSupport {
-    
+
     protected ApplicationXmlDocument applicationXml = new ApplicationXmlDocument();
-    
-    protected void installApp(Artifact artifact) throws Exception {
-        
-        if (artifact.getFile() == null || artifact.getFile().isDirectory()) {
+
+    protected void installApp(Artifact artifact, boolean thin) throws Exception {
+        File destDir = new File(serverDirectory, getAppsDirectory());
+
+        if (artifact.getFile() == null || artifact.getFile().isDirectory() || thin) {
             String warName = getAppFileName(project);
+            if (thin) {
+                warName = "thin-" + warName;
+                // When the apps directory is "dropins" place the thin application in
+                // dropins/spring folder and
+                // when the apps directory is "apps" place the thin application in apps folder
+                if (destDir.getName().equalsIgnoreCase("dropins")) {
+                    destDir = new File(destDir, "spring");
+                }
+                copyLibIndexCache();
```

Comment:
If the thin app doesn't exist, an exception will get thrown at line 61, but we will have copied over the lib.index.cache file already. Wondering if this copy should be done later in this method?

Reply:
Thrown an exception in copyLibIndexCache().